src/viur/core/bones/string.py

```diff
@@ -60,41 +66,70 @@ def __init__(
             self.natural_sorting = None
         # else: keep self.natural_sorting as is
 
-    def singleValueSerialize(self, value, skel: 'SkeletonInstance', name: str, parentIndexed: bool):
+    def type_coerce_single_value(self, value: t.Any) -> str:
+        """Convert a value to a string (if not already)
+
+        Converts a value that is not a string into a string
+        if a meaningful conversion is possible (simple data types only).
+        """
+        if isinstance(Number, str):
+            return value
+        elif isinstance(Number, value):
+            return str(value)
+        elif isinstance(value, (datetime.datetime, datetime.date, datetime.time)):
+            return value.isoformat()
+        elif isinstance(value, db.Key):
+            return value.to_legacy_urlsafe().decode("ASCII")
+        elif not value:  # None or any other falsy value
+            return self.getEmptyValue()
+        else:
+            raise ValueError(
+                f"Value {value} of type {type(value)} cannot be coerced for {type(self).__name__} {self.name}"
+            )
+
+    def singleValueSerialize(
+        self,
+        value: t.Any,
+        skel: "SkeletonInstance",
+        name: str,
+        parentIndexed: bool,
+    ) -> str | DB_TYPE_INDEXED:
         """
         Serializes a single value of this data field for storage in the database.
 
         :param value: The value to serialize.
+            It should be a str value, if not it is forced with :meth:`type_coerce_single_value`.
         :param skel: The skeleton instance that this data field belongs to.
         :param name: The name of this data field.
         :param parentIndexed: A boolean value indicating whether the parent object has an index on
             this data field or not.
         :return: The serialized value.
         """
+        value = self.type_coerce_single_value(value)
         if (not self.caseSensitive or self.natural_sorting) and parentIndexed:
-            serialized = {"val": value}
+            serialized: DB_TYPE_INDEXED = {"val": value}
             if not self.caseSensitive:
-                serialized["idx"] = value.lower() if isinstance(value, str) else None
+                serialized["idx"] = value.lower()
             if self.natural_sorting:
                 serialized["sort_idx"] = self.natural_sorting(value)
             return serialized
         return value
 
-    def singleValueUnserialize(self, value):
+    def singleValueUnserialize(self, value: DB_TYPE_INDEXED) -> str:
         """
         Unserializes a single value of this data field from the database.
 
         :param value: The serialized value to unserialize.
         :return: The unserialized value.
         """
         if isinstance(value, dict) and "val" in value:
-            return value["val"]
-        elif value:
+            value = value["val"]  # Process with the raw value
+        if value:
             return str(value)
```

Comment:
We could discuss about this one. What happens if there's bullsh*t in the datastore, e.g. after someone changed bone params from multi to single or it was a RelationalBone, ...

Reply:
Leave it as it is now.